src/lib.rs

```diff
@@ -1552,8 +1558,8 @@ where
     {
         if let Some(slc) = self.as_slice_memory_order_mut() {
             // FIXME: Use for loop when slice iterator is perf is restored
-            for i in 0..slc.len() {
-                f(&mut slc[i]);
+            for x in slc.iter_mut() {
```

Comment:
Are the performance issues solved now? This comment was from several years ago

Reply:
The commit adding the comment (86415f780b6ef70be4eb314fab36c3230ee6ac29) says

> Use counted loop in unordered_foreach_mut
>
> This improves the generated code for the inner loop we use for
> .assign_scalar() and other array operations with a single scalar.
>
> The old code would generate something that looked like it assumed the
> element being written to and the iterator's pointer could alias another. This
> new version is unrolled and vectorized to a larger degree, but it does
> still not produce memset for some inputs like rust could do in january 2016.
>
> name                     before ns/iter  after ns/iter    diff ns/iter
> diff %
> assign_scalar_2d_corder  1,631           868                      -763 -46.78%
> assign_scalar_2d_cutout  1,704           1,332                    -372 -21.83%
> assign_scalar_2d_forder  1,628           869                      -759 -46.62%
> assign_zero_2d_corder    1,602           855                      -747 -46.63%
> assign_zero_2d_cutout    1,703           1,335                    -368 -21.61%
> assign_zero_2d_forder    1,618           858                      -760 -46.97%
>
> before: Using the slice iterator
> after: using indexed loop.

I tried switching the implementation to use `for_each` and got the same results for these benchmarks as the range-based implementation, so I think that would work well. (I didn't try a `for` loop.)

```rust
        if let Some(slc) = self.as_slice_memory_order_mut() {
            slc.iter_mut().for_each(f);
        } else {
            for row in self.inner_rows_mut() {
                row.into_iter_().fold((), |(), elt| f(elt));
            }
        }
```